clang/lib/CIR/CodeGen/CIRGenBuiltinX86.cpp

```diff
@@ -1706,7 +1706,12 @@ CIRGenFunction::emitX86BuiltinExpr(unsigned builtinID, const CallExpr *expr) {
                                     cir::SyncScopeKind::SingleThread));
     return mlir::Value{};
   }
-  case X86::BI_AddressOfReturnAddress:
+  case X86::BI_AddressOfReturnAddress: {
+    mlir::Location loc = getLoc(expr->getExprLoc());
+    mlir::Value addr = emitIntrinsicCallOp(
+        builder, loc, "addressofreturnaddress", allocaInt8PtrTy);
```

Comment:
We have a CIR operation to get the return address (`CIR_ReturnAddrOp`) so we should probably add an operation for this also. Look at https://github.com/llvm/llvm-project/pull/153698 for an example.